src/test/java/com/connexta/search/index/IndexServiceImplTest.java

```diff
@@ -6,239 +6,175 @@
-    when(mockRequestHeadersSpec.retrieve()).thenReturn(mockResponseSpec);
-    when(mockResponseSpec.onStatus(any(Predicate.class), any(Function.class)))
-        .thenReturn(mockResponseSpec2);
-    when(mockResponseSpec2.bodyToMono(Resource.class)).thenReturn(mockMono);
-    when(mockMono.block()).thenReturn(mockResource);
 
-    final IOException ioException = new IOException();
-    when(mockResource.getInputStream()).thenThrow(ioException);
+    // and stub dataset already exists
+    final RuntimeException runtimeException = new RuntimeException();
+    doThrow(runtimeException).when(mockIndexRepository).existsById(datasetId);
 
+    // expect
     final SearchException thrown =
         assertThrows(
-            SearchException.class,
-            () ->
-                ((IndexService) new IndexServiceImpl(mockSearchManager, mockWebClient))
-                    .index(datasetId, irmUri));
-    assertThat(thrown.getStatus(), is(HttpStatus.BAD_REQUEST));
-    assertThat(thrown.getCause(), is(ioException));
+            SearchException.class, () -> indexService.index(Index.builder().id(datasetId).build()));
+    assertThat(thrown.getCause(), is(runtimeException));
   }
 
-  @ParameterizedTest
-  @MethodSource("exceptionsThrownSearchManager")
-  void testSearchManagerExceptions(final RuntimeException runtimeException) throws Exception {
+  @Test
+  void testExceptionWhenSaving() throws Exception {
+    // given
     final String datasetId = "00067360b70e4acfab561fe593ad3f7a";
+    final String contents =
+        "All the color had been leached from Winterfell until only grey and white remained.";
 
-    final RuntimeException thrown;
-
-    try (final MockWebServer storeMockWebServer = new MockWebServer()) {
-      storeMockWebServer.start();
-      storeMockWebServer.enqueue(
-          new MockResponse()
-              .setBody(
-                  String.format(
-                      "{ \"%s\" : \"All the color had been leached from Winterfell until only grey and white remained.\" }",
-                      EXT_EXTRACTED_TEXT))
-              .setResponseCode(200));
-      final URI irmUri = storeMockWebServer.url(String.format("/dataset/%s/irm", datasetId)).uri();
-      doThrow(runtimeException)
-          .when(mockSearchManager)
-          .index(eq(datasetId), eq(irmUri), any(InputStream.class));
-
-      thrown =
-          assertThrows(
-              RuntimeException.class,
-              () ->
-                  ((IndexService) new IndexServiceImpl(mockSearchManager, WebClient.create()))
-                      .index(datasetId, irmUri));
-    }
-
-    assertThat(
-        "thrown exception is the exact same exception thrown by the SearchManager",
-        thrown,
-        is(runtimeException));
+    final URI irmUri = new URI(String.format("http://store:9041/dataset/%s/irm", datasetId));
+    final URI fileUri = new URI(String.format("http://store:9041/dataset/%s/", datasetId));
```

Comment:
✏️ No need to construct URIs and then covert to strings. These can also be inlined.